src/modules/control/src/lib_high_level_control.cpp

```diff
@@ -0,0 +1,282 @@
+#include "control/lib_high_level_control.h"
+
+namespace fusionad
+{
+namespace control
+{
+namespace libraries
+{
+
+  Waypoints::Waypoints()
+  {
+  }
+
+  Waypoints::~Waypoints()
+  {
+  }
+
+  bool Waypoints::SetPath(const nav_msgs::Path& incoming_path_list)
+  {
+    current_path_list_.poses.clear();
+    current_path_list_ = incoming_path_list;
+    // Only perform path stuffing if the current path data is empty
+    if(!incoming_path_list.poses.empty())
+    {
+      for(int i = 0; i < GetPathSize(); i++)
+      {
+        tf2::Quaternion waypoint_quaternion;
+        float waypoint_theta = 0;
+        try
+        {
+          waypoint_theta = ComputeWaypointTheta(i);
+        }
+        catch(const char* msg)
+        {
+          return false;
+        }
+        waypoint_quaternion.setRPY(0,0,waypoint_theta);
+        tf2::convert(waypoint_quaternion,current_path_list_.poses[i].pose.orientation);
+      }
+      return true;
+    }
+    else
+    {
+      throw "Incoming path empty, path is not set!";
+    }
+  }
+
+  float Waypoints::GetWaypointTheta(const int& index)
+  {
+    if(!IsPathEmpty())
+    {
+      double waypoint_roll, waypoint_pitch, waypoint_yaw;
+      tf2::Quaternion waypoint_quaternion;
+      tf2::convert(current_path_list_.poses[index].pose.orientation, waypoint_quaternion);
+      tf2::Matrix3x3 waypoint_quaternion_matrix(waypoint_quaternion);
+      waypoint_quaternion_matrix.getRPY(waypoint_roll, waypoint_pitch, waypoint_yaw);
+      float waypoint_theta = static_cast<float>(waypoint_yaw);
+      if(!std::isfinite(waypoint_theta))
+      {
+        throw "Waypoint Theta not finite!";
+      }
+      else
+      {
+        return waypoint_theta;
+      }
+    }
+    else
+    {
+      throw "Path is not set!";
```

Comment:
I do not plan to implement any ROS main's feature (ROS_log) here since this is intended to be a standalone library that has ROS' message compatibility. The log and other higher level error checking features will be done on the "node-side" logic for the other PR.